syntax = "proto3";

service StringFormatter {
    rpc Upper (UpperRequest) returns (UpperReply); // makes all chars in string upper case 
    rpc Lower (LowerRequest) returns (LowerReply); // makes all chars in string lower case
    rpc Proper (ProperRequest) returns (ProperReply); // makes first char in every word upper case, all other chars lower case
}

// RPC Upper
message UpperRequest{
    string upperReqStr = 1;
}
message UpperReply{
    string upperRepStr = 1;
}

// RPC Lower
message LowerRequest{
    string lowerReqStr = 1;
}
message LowerReply{
    string lowerRepStr = 1;
}

// RPC Proper
message ProperRequest{
    string propReqStr = 1;
}
message ProperReply{
    string  propRepStr = 1;
}

//COMMAND LINE TO COMPILE: python -m grpc_tools.protoc -I../../protos --python_out=. --grpc_python_out=. ../../protos/route_guide.proto